### PR TITLE
String and number path param validation

### PR DESCRIPTION
**Pull request by emilien-puget**
this PR depends on https://github.com/pb33f/libopenapi-validator/pull/55 for the validateSimpleParam function

it should fix https://github.com/pb33f/libopenapi-validator/issues/43

### CODE REVIEW

**Comment by emilien-puget** [comment]
I'll add the test to cover that case

File: parameters/path_parameters.go [hunk 1]
```diff
@@ -280,3 +281,27 @@ func (v *paramValidator) ValidatePathParams(request *http.Request) (bool, []*err
...
+	if isMatrix && p.Style == helpers.MatrixStyle {
+		// strip off the colon and the parameter name
+		paramValue = strings.Replace(paramValue[1:], fmt.Sprintf("%s=", p.Name), "", 1)
+		paramValueParsed, err := strconv.ParseFloat(paramValue, 64)
+		if err != nil {
+			return "", 0, []*errors.ValidationError{errors.IncorrectPathParamNumber(p, paramValue[1:], sch)}
+		}
+		return paramValue, paramValueParsed, nil
+	}
+	paramValueParsed, err := strconv.ParseFloat(paramValue, 64)
```
Comment (emilien-puget) [inline parameters/path_parameters.go:302]: this is already parsed and checked there https://github.com/emilien-puget/libopenapi-validator/blob/c16b8e7fd4972fcf7c7beb2c612857f7cd910927/paths/paths.go#L291, should we validate during the path resolve for simple case ? (not object) @daveshanley 
the error could be path not found instead too

Comment (emilien-puget) [reply parameters/path_parameters.go:302]: I am wondering if path parameters validation must take place during path matching with the request or after, I'll take a look on how it is done in other system

Comment (emilien-puget) [reply parameters/path_parameters.go:302]: @daveshanley i'd suggest to remove the path param validation during the find path process and only validate the parameters when validating the request, wdyt ?

**Comment by emilien-puget** [comment]
@daveshanley i pushed the proposed changes, with those changes we got more precise errors when a bad path parameter is given during a validation context

**Comment by daveshanley** [comment]
Would you be able to rebase?

File: parameters/path_parameters.go (hunk 1 above)
Comment (daveshanley) [reply parameters/path_parameters.go:302]: OK, sounds good.

**Comment by emilien-puget** [comment]
> Would you be able to rebase?

Of course, I'll do that

**Comment by emilien-puget** [comment]
@daveshanley rebased and squashed

**Pull request merged**